components/cache/index.js

```diff
@@ -0,0 +1,427 @@
+		);
+		if (!this.isReady()) {
+			return cb(errorCacheDisabled);
+		}
+		const pattern = CACHE.KEYS.delegatesApi;
+		return this.removeByPattern(pattern, err => {
+			if (err) {
+				this.logger.error(
+					[
+						'Cache - Error clearing keys with pattern:',
+						pattern,
+						'round finish',
+					].join(' ')
+				);
+			} else {
+				this.logger.debug(
+					[
+						'Cache - Keys with pattern:',
+						pattern,
+						'cleared from cache on new Round',
+					].join(' ')
+				);
+			}
+			return cb(err);
+		});
+	}
+
+	/**
+	 * Clears all cache entries if there is a delegate type transaction after transactions saved.
+	 *
+	 * @param {transactions[]} transactions
+	 * @param {function} cb
+	 * @todo Add description for the params
+	 * @todo Add @returns tag
+	 */
+	onTransactionsSaved(transactions, cb) {
+		cb = cb || function() {};
+
+		this.logger.debug(
+			['Cache - onTransactionsSaved', '| Status:', this.isConnected()].join(' ')
+		);
+		if (!this.isReady()) {
+			return cb(errorCacheDisabled);
+		}
+
+		return async.parallel(
+			[
+				async.reflect(reflectCb => {
+					const pattern = CACHE.KEYS.delegatesApi;
+
+					const delegateTransaction = transactions.find(
+						transaction =>
+							!!transaction && transaction.type === transactionTypes.DELEGATE
+					);
+
+					if (!delegateTransaction) {
+						return setImmediate(reflectCb, null);
+					}
+
+					return this.removeByPattern(pattern, removeByPatternErr => {
+						if (removeByPatternErr) {
+							this.logger.error(
+								[
+									'Cache - Error clearing keys with pattern:',
+									pattern,
+									'on delegate transaction',
+								].join(' ')
+							);
+						} else {
+							this.logger.debug(
+								[
+									'Cache - Keys with pattern:',
+									pattern,
+									'cleared from cache on delegate transaction',
+								].join(' ')
+							);
+						}
+						return setImmediate(reflectCb, removeByPatternErr);
+					});
+				}),
+
+				async.reflect(reflectCb => {
+					if (transactions.length === 0) {
+						return setImmediate(reflectCb, null);
+					}
+
+					return this.deleteJsonForKey(
+						CACHE.KEYS.transactionCount,
+						deleteJsonForKeyErr => {
+							if (deleteJsonForKeyErr) {
+								this.logger.error(
+									`Cache - Error clearing keys: ${CACHE.KEYS.transactionCount}`
+								);
+							} else {
+								this.logger.debug(
+									`Cache - Keys ${
+										CACHE.KEYS.transactionCount
+									} cleared from cache`
+								);
+							}
+							return setImmediate(reflectCb, deleteJsonForKeyErr);
+						}
+					);
+				}),
+			],
+			() => setImmediate(cb, null) // Don't propagate cache error to continue normal operations
+		);
+	}
+
+	/**
+	 * Disables any changes in cache while syncing.
+	 */
+	onSyncStarted() {
+		this.cacheReady = false;
+	}
+
+	/**
+	 * Enables changes in cache after syncing finished.
+	 */
+	onSyncFinished() {
+		this.cacheReady = true;
+	}
```

Comment:
In principal no events should be part of the component. Consider component to be just a shared library, so activities on events should move to the their respective namespaces. 

Consider this in relation, what if we use this cache component in the `http-api`, there are no such events there, it just need a wrapper to get and set the cache. And cache keys to be there, so both modules can reply on same keys.